modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectFieldFormBase.tsx

```diff
@@ -434,6 +528,193 @@ export function useObjectFieldForm({
 	return {errors, handleChange, handleSubmit, setValues, values};
 }
 
+function AggregationSourceProperty({
+	disabled,
+	errors,
+	objectRelationships,
+	objectFieldSettings = [],
+	setValues,
+}: IAggregationSourcePropertyProps) {
+	const [query, setQuery] = useState<string>('');
+	const [selectedRelatedObject, setSelectRelatedObject] = useState<
+		TObjectRelationship
+	>();
+	const [selectedSummarizeField, setSelectedSummarizeField] = useState<
+		string
+	>();
+	const [
+		selectedAggregationFunction,
+		setSelectedAggregationFunction,
+	] = useState<{label: string; value: string}>();
+	const [objectRelationshipFields, setObjectRelationshipFields] = useState<
+		ObjectField[]
+	>();
+
+	const handleChangeRelatedObject = async (
+		objectRelationship: TObjectRelationship
+	) => {
+		setSelectRelatedObject(objectRelationship);
+		setSelectedSummarizeField('');
+
+		const relatedFields = await fetchObjectFields(
+			objectRelationship.objectDefinitionId2
+		);
+
+		const numericFields = relatedFields.filter(
+			(objectField) =>
+				objectField.businessType === 'Integer' ||
+				objectField.businessType === 'LongInteger' ||
+				objectField.businessType === 'Decimal' ||
+				objectField.businessType === 'PrecisionDecimal'
+		);
+
+		setObjectRelationshipFields(numericFields);
+
+		const fieldSettingWithoutSummarizeField = objectFieldSettings.filter(
+			(fieldSettings) => fieldSettings.name !== 'summarizeField'
+		);
+
+		const newObjectFieldSettings: ObjectFieldSetting[] | undefined = [
+			...fieldSettingWithoutSummarizeField.filter(
+				(fieldSettings) => fieldSettings.name !== 'relatedObject'
+			),
+			{
+				name: 'relatedObject',
+				value: objectRelationship.name,
+			},
+		];
+
+		setValues({
+			objectFieldSettings: newObjectFieldSettings,
+		});
+	};
+
+	const handleAggregationFuncionChange = (value: {
+		label: string;
+		value: string;
```

Comment:
@danvb112 To avoid cases like the call bellow `value.value` you could name this value as `key` (we would have `value.key)` or something similar. What do you think?

Reply:
Hi @naraandrade, I can't rename this name from prop to `key`, as it comes from the `FormCustomSelect` component with that name, but to avoid using `value.value` I can unstructure the parameter and get only the `label` and the `value` , what do you think?

Reply:
@danvb112 I think it's a good idea for this case :) The goal is to make the code more clear.